src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java

```diff
@@ -108,12 +109,35 @@ private CategorizedClinicalDataCountFilter extractClinicalDataCountFilters(final
     }
 
     private void buildClinicalAttributeNameMap() {
+        List<ClinicalAttribute> clinicalAttributes = studyViewRepository.getClinicalAttributes();
         List<ClinicalAttributeDataSource> clinicalAttributeDataSources = List.of(ClinicalAttributeDataSource.values());
+        // TODO we can probably move this entire group by datatype and isPatientAttr logic to SQL
         for (ClinicalAttributeDataSource clinicalAttributeDataSource : clinicalAttributeDataSources) {
             String categoricalKey = clinicalAttributeDataSource.getValue() + ClinicalAttributeDataType.CATEGORICAL;
             String numericKey = clinicalAttributeDataSource.getValue() + ClinicalAttributeDataType.NUMERIC;
-            clinicalAttributeNameMap.put(categoricalKey, studyViewRepository.getClinicalDataAttributeNames(clinicalAttributeDataSource, ClinicalAttributeDataType.CATEGORICAL));
-            clinicalAttributeNameMap.put(numericKey, studyViewRepository.getClinicalDataAttributeNames(clinicalAttributeDataSource, ClinicalAttributeDataType.NUMERIC));
+            boolean isPatientAttr = clinicalAttributeDataSource.equals(ClinicalAttributeDataSource.PATIENT);
+            clinicalAttributeNameMap.put(
+                categoricalKey, 
+                clinicalAttributes
+                    .stream()
+                    .filter(a -> 
+                        a.getDatatype().equals(ClinicalAttributeDataType.CATEGORICAL.getValue()) &&
+                            isPatientAttr == a.getPatientAttribute()
+                    )
+                    .map(ClinicalAttribute::getAttrId)
+                    .toList()
+            );
+            clinicalAttributeNameMap.put(
+                numericKey,
+                clinicalAttributes
+                    .stream()
+                    .filter(a -> 
+                        a.getDatatype().equals(ClinicalAttributeDataType.NUMERIC.getValue()) && 
+                            isPatientAttr == a.getPatientAttribute()
+                    )
+                    .map(ClinicalAttribute::getAttrId)
+                    .toList()
+            );
```

Comment:
I would say all this not needed anymore with your call to studyViewRepository.getClinicalAttributes() now.

We originally needed this because there were 4 separate tables for clinicalAttribute.

Now CategorizedClinicalDataFilter can be pushed down into the repository layer and have one call that grabs all clinicalAttribute_meta data. Then we can split patient and sample filtering